### ISSUE TITLE
B023: false positive for nested function inside loop

### ISSUE THREAD

**Issue by fwinkl** [PyCQA/flake8-bugbear#468]
The following (admittedly bad) code produces what I think must be a false positive:
```py
for _ in range(10):
   foo = []
   def bar():
      foo.append(42)

   bar()
```
```
test.py:4:7: B023 Function definition does not bind loop variable 'foo'.
```

At the very least the warning is confusing because `foo` is not actually a loop variable.

Using:
```
flake8 --version
7.0.0 (flake8-bugbear: 24.2.6, flake8-builtins: 2.5.0, mccabe: 0.7.0, pycodestyle: 2.11.1, pyflakes: 3.2.0) CPython 3.10.12 on Linux
```

**Comment by cooperlees** [comment]
Agree. I think we should ignore this. I don't think this is particularly worth a dedicated check here, but could be convinced.
